junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/ParameterExtensionTest.java

```diff
@@ -126,167 +118,177 @@ public void producerConfiguration(KafkaCluster cluster,
-        assertNotEquals(cluster1.getClusterId(), cluster2.getClusterId());
-        var dc1 = describeCluster(cluster1.getKafkaClientConfiguration());
-        assertEquals(cluster1.getClusterId(), dc1.clusterId().get());
-        assertEquals(1, dc1.nodes().get().size());
-        var dc2 = describeCluster(cluster2.getKafkaClientConfiguration());
-        assertEquals(cluster2.getClusterId(), dc2.clusterId().get());
-        await().atMost(CLUSTER_FORMATION_TIMEOUT).untilAsserted(() -> assertEquals(2, describeCluster(cluster2.getKafkaClientConfiguration()).nodes().get().size()));
-    }
+    void twoAnonClusterParameter(
+                                 @BrokerCluster(numBrokers = 1) KafkaCluster cluster1,
+                                 @BrokerCluster(numBrokers = 2) KafkaCluster cluster2)
+            throws Exception {
+        assertThat(cluster1.getClusterId()).isNotEqualTo(cluster2.getClusterId());
 
-    // @Name is not required here because there's no ambiguity
-    @Test
-    public void twoDefinedClusterParameter(
-                                           @BrokerCluster(numBrokers = 1) KafkaCluster cluster1,
-                                           @BrokerCluster(numBrokers = 2) KafkaCluster cluster2)
-            throws ExecutionException, InterruptedException {
-        assertEquals(1, describeCluster(cluster1.getKafkaClientConfiguration()).nodes().get().size());
-        await().atMost(CLUSTER_FORMATION_TIMEOUT).untilAsserted(() -> assertEquals(2, describeCluster(cluster2.getKafkaClientConfiguration()).nodes().get().size()));
+        assertClusterIdAndSize(cluster1, 1);
+        assertClusterIdAndSize(cluster2, 2);
     }
 
     @Test
-    public void twoDefinedClusterParameterAndAdmin(
-                                                   @BrokerCluster(numBrokers = 1) @Name("A") KafkaCluster clusterA,
-                                                   @BrokerCluster(numBrokers = 2) @Name("B") KafkaCluster clusterB,
-                                                   @Name("B") Admin adminB,
-                                                   @Name("A") Admin adminA)
-            throws ExecutionException, InterruptedException {
+    void twoDefinedClusterParameterAndAdmin(
+                                            @BrokerCluster(numBrokers = 1) @Name("A") KafkaCluster clusterA,
+                                            @BrokerCluster(numBrokers = 2) @Name("B") KafkaCluster clusterB,
+                                            @Name("B") Admin adminB,
+                                            @Name("A") Admin adminA)
+            throws Exception {
         assertSameCluster(clusterA, adminA);
-        assertEquals(1, describeCluster(clusterA.getKafkaClientConfiguration()).nodes().get().size());
-        assertEquals(1, describeCluster(adminA).nodes().get().size());
+        assertClusterIdAndSize(clusterA, 1);
+
         assertSameCluster(clusterB, adminB);
```

Comment:
Ohh should we be also have `assertSameCluster(clusterA, adminA);`?

Reply:
`assertSameCluster(clusterA, adminA);` is already part of this test (twoDefinedClusterParameterAndAdmin)